Admin Page UI components

CITIZENDOT opened the pull request:
## Summary
![image](https://github.com/EtherealEngine/etherealengine/assets/52322531/cef590e9-44fd-40da-a356-6d24bca75beb)

![image](https://github.com/EtherealEngine/etherealengine/assets/55396651/018d4536-7998-4b21-a34c-b68b9487f171)

## References
closes #_insert number here_

## QA Steps

CITIZENDOT committed:
Add Table Component

CITIZENDOT committed:
adjust spacing in story

aditya-mitra committed:
Merge branch 'dev' into tailwind-table

aditya-mitra committed:
put classes directly into tailwind table

aditya-mitra committed:
add table test

aditya-mitra committed:
added `xadmin` route for accessing the new admin panel

aditya-mitra committed:
added surface main theme

aditya-mitra committed:
improvements in tailwind table

CITIZENDOT committed:
Merge branch 'dev' into tailwind-table

CITIZENDOT committed:
Improve modal styles

CITIZENDOT committed:
Add Callout Text component

CITIZENDOT committed:
Merge branch 'dev' into tailwind-table

aditya-mitra committed:
Merge branch 'dev' into tailwind-table

aditya-mitra committed:
remove engine canvas and fix table style

CITIZENDOT committed:
Use popover state for modal

CITIZENDOT committed:
Merge branch 'dev' into tailwind-table

CITIZENDOT committed:
fix conflicts

CITIZENDOT committed:
Added copy text component

aditya-mitra committed:
added project table data and fix pagination

aditya-mitra committed:
Merge branch 'tailwind-table-1' into tailwind-table

aditya-mitra committed:
Merge branch 'dev' into tailwind-table

aditya-mitra committed:
added confirm dialog and fix border styles

aditya-mitra committed:
added AddEditProjectModal dialog and Label component

aditya-mitra committed:
added layout for update engine

aditya-mitra committed:
Merge branch 'dev' into tailwind-table

aditya-mitra committed:
added checkbox, select components

wire up UpdateEngineModal

aditya-mitra committed:
added functionalities for AddProject

aditya-mitra committed:
Merge branch 'dev' into tailwind-table

aditya-mitra committed:
add project submit

aditya-mitra committed:
add multiple modal support

aditya-mitra committed:
update project functionality

aditya-mitra committed:
correct multiple modal rendering behavior

aditya-mitra committed:
Merge branch 'dev' into tailwind-table

aditya-mitra committed:
added toggle tailwind

aditya-mitra committed:
added user permission modal

CITIZENDOT committed:
Add scrollbar to table

CITIZENDOT committed:
Merge branch 'tailwind-table' of github.com:EtherealEngine/etherealengine into tailwind-table

HexaField commented on packages/client-core/src/common/services/PopoverState.tsx:
```diff
@@ -23,14 +23,26 @@ All portions of the code written by the Ethereal Engine team are Copyright © 20
+
+  /**shows a popupover. if a previous popover was already present, the `element` popover will be current showed */
+  showPopupover: (element: JSX.Element) => {
+    getMutableState(PopoverState).elements.set((prevElements) => [...prevElements, element])
```
this can be simplified to .merge([element])

CITIZENDOT committed:
Add tooltip component

tailwind doesnot seem to support nested pseudo element in styles, for eg: `hover:before:`. Hence, raw CSS is used here. TODO: to be updated with alternative implementation when found.

CITIZENDOT committed:
Show loading indicator when projects are being loaded

CITIZENDOT committed:
Improve Sidebar styles

CITIZENDOT committed:
Merge branch 'dev' into tailwind-table

aditya-mitra committed:
regression on loading spinner

aditya-mitra committed:
additional styles to table to accomodate max width
